wsagent/che-datasource-api/src/main/java/org/eclipse/che/datasource/api/ssl/KeyStoreObject.java

```diff
@@ -0,0 +1,184 @@
+            ks.load(fis, keyStorePassword.toCharArray());
+        } catch (Exception e) {
+            LOG.info("Couldn't load keystore file ");
+            ks.load(null, keyStorePassword.toCharArray());
+        }
+
+        return ks;
+    }
+
+
+    protected String getKeyStorePreferenceName() {
+        return SSL_KEY_STORE_PREF_ID;
+    }
+
+//    private String getUserId() {
+//        return EnvironmentContext.getCurrent().;
+//    }
+
+    protected String getKeyStorePassword() {
+        return SslKeyStoreService.getDefaultKeystorePassword();
+    }
+
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response getKeyList(@Context UriInfo uriInfo) throws Exception {
+        List<SslKeyStoreEntry> result = new ArrayList<SslKeyStoreEntry>(keystore.size());
+
+        Enumeration<String> e = keystore.aliases();
+        while (e.hasMoreElements()) {
+            String alias = e.nextElement();
+            SslKeyStoreEntry sslKeyStoreEntry = DtoFactory.getInstance().createDto(SslKeyStoreEntry.class)
+                    .withAlias(alias)
+                    .withType(
+                            keystore.isKeyEntry(alias) ? "Key" : "Certificate"
+                    );
+            result.add(sslKeyStoreEntry);
+        }
+        return Response.ok().entity(result).type(MediaType.APPLICATION_JSON).build();
+    }
+
+    @POST
+    @Path("add")
+    @Consumes({MediaType.MULTIPART_FORM_DATA})
+    public Response addNewKeyCertificateAndRespond(@QueryParam("alias") String alias,
+                                                   Iterator<FileItem> uploadedFilesIterator) throws Exception {
+        addNewKey(alias, uploadedFilesIterator);
+        return Response.ok("", MediaType.TEXT_HTML).build();
+    }
+
+    public void addNewKey(String alias, Iterator<FileItem> uploadedFilesIterator) throws Exception {
+        PrivateKey privateKey = null;
+        Certificate[] certs = null;
+        while (uploadedFilesIterator.hasNext()) {
+            FileItem fileItem = uploadedFilesIterator.next();
+            if (!fileItem.isFormField()) {
+                if ("keyFile".equals(fileItem.getFieldName())) {
+                    KeyFactory kf = KeyFactory.getInstance("RSA");
+                    privateKey = kf.generatePrivate(new PKCS8EncodedKeySpec(fileItem.get()));
+                }
+                if ("certFile".equals(fileItem.getFieldName())) {
+                    CertificateFactory cf = CertificateFactory.getInstance("X.509");
+                    certs = cf.generateCertificates(fileItem.getInputStream()).toArray(new Certificate[]{});
+                }
+            }
+        }
+
+        if (privateKey == null || certs == null) {
+            throw new WebApplicationException(Response.ok("<pre>Can't find input file.</pre>", MediaType.TEXT_HTML).build());
+        }
+
+        keystore.setKeyEntry(alias, privateKey, keyStorePassword.toCharArray(), certs);
```

Comment:
I see that in some places you call it `keystore` while in other - `keyStore` it would be very appreciated if you would choose the only one way of naming that identifier. What's your opinion, does that make sense to you?